python/ikarus/io/__init__.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+# SPDX-License-Identifier: LGPL-3.0-or-later
+
+from dune.common.hashit import hashIt
+from ikarus.generator import MySimpleGenerator
+from dune.vtk import FormatTypes, DataTypes, vtkWriter
+import dune.vtk
+
+from warnings import warn
+from io import StringIO
+from dune.generator.algorithm import run
+import types
+
+# The list of supported dataCollectors
+dataCollectors = ["lagrange", "discontinuous", "iga"]
+
+
+def __addInterpolation(writer, flag: int):
+    typeStr = "Ikarus::Vtk::asPointData" if flag == 1 else "Ikarus::Vtk::asCellData"
+
+    def __addInterpolationFunc(writer, vals_, basis, name: str, size: int):
+        containerStr = f"Dune::FieldVector<double, {size}>" if size > 1 else "double"
+
+        runCode = """
+            #define EIGEN_DEFAULT_TO_ROW_MAJOR 1
+            #include <ikarus/python/io/vtkwriter.hh>
+            #include <dune/python/pybind11/eigen.h>
+            template <typename Writer, typename Basis>
+            void addInterpolation(Writer& writer, const auto& vals_, const Basis& basis, std::string name) {{
+                auto vals = vals_.template cast<Eigen::VectorX<double>>();
+                writer.template addInterpolation<Basis, {containerStr}>(std::move(vals), basis, name, {typeStr});
```

Comment:
This is not needed with my suggested modifications and this can be then  be a normal member function of the vtkwriter defined above.

Reply:
I think its still necessary, at least when we want to support both the global basis and any kind of subspace basis right?